hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -875,6 +916,86 @@ ChangesetInfoPtr XmlChangeset::splitChangeset(ChangesetInfoPtr changeset)
     updateFailedChangeset(changeset);
     return split;
   }
+  //  Try to use the split hint to split out one failing element
+  if (splitHint != "")
+  {
+    long member_id = 0;
+    long element_id = 0;
+    ElementType::Type member_type = ElementType::Unknown;
+    ElementType::Type element_type = ElementType::Unknown;
+    //  See if the hint is something like: Placeholder node not found for reference -145213 in way -5687
+    if (matchesPlaceholderFailure(splitHint, member_id, member_type, element_id, element_type))
+    {
+      //  Use the type and id to split the changeset
+      if (element_type == ElementType::Way)
+      {
+        for (int current_type = ChangesetType::TypeCreate; current_type != ChangesetType::TypeMax; ++current_type)
+        {
+          if (changeset->contains(element_type, (ChangesetType)current_type, element_id))
+          {
+            XmlWay* way = dynamic_cast<XmlWay*>(_allWays[element_id].get());
+            //  Add the way to the split and remove from the changeset
+            split->add(element_type, (ChangesetType)current_type, way->id());
+            changeset->remove(element_type, (ChangesetType)current_type, way->id());
+            return split;
+          }
+        }
+      }
+      else if (element_type == ElementType::Relation)
+      {
+        for (int current_type = ChangesetType::TypeCreate; current_type != ChangesetType::TypeMax; ++current_type)
+        {
+          if (changeset->contains(element_type, (ChangesetType)current_type, element_id))
+          {
+            XmlRelation* relation = dynamic_cast<XmlRelation*>(_allRelations[element_id].get());
+            //  Add the relation to the split and remove from the changeset
+            split->add(element_type, (ChangesetType)current_type, relation->id());
+            changeset->remove(element_type, (ChangesetType)current_type, relation->id());
+            return split;
+          }
+        }
+      }
+    }
+    //  See if the hint is something like: Relation with id  cannot be saved due to Relation with id 1707699
+    if (matchesRelationFailure(splitHint, element_id, member_id, member_type))
+    {
+      if (element_id != 0)
+      {
+        //  If there is a relation id, move just that relation to the split
+        for (int current_type = ChangesetType::TypeCreate; current_type != ChangesetType::TypeMax; ++current_type)
+        {
+          if (changeset->contains(member_type, (ChangesetType)current_type, element_id))
+          {
+            XmlRelation* relation = dynamic_cast<XmlRelation*>(_allRelations[element_id].get());
+            //  Add the relation to the split and remove from the changeset
+            split->add(ElementType::Relation, (ChangesetType)current_type, relation->id());
+            changeset->remove(ElementType::Relation, (ChangesetType)current_type, relation->id());
+            return split;
+          }
```

Comment:
In this scenario the relation that failed has an ID that comes back from the API.  It is parsed and moved to the split and failure state.